go.mod

```diff
@@ -20,6 +20,7 @@ require (
 	github.com/olekukonko/tablewriter v0.0.4
 	github.com/open-policy-agent/opa v0.24.0
 	github.com/opencontainers/image-spec v1.0.1
+	github.com/shteou/go-ignore v0.3.0
```

Comment:
It's worth drawing attention to this library. I figured it didn't make sense to include the code to parse in conftest (as far as I can tell, it relies on other libraries to do this in all cases), so I wrote a library for this which did nothing but parsing, to an exported struct. I'm not wedded to it, and happy to replace with an existing library.

There were two other golang gitignore parsers I could find:
https://github.com/monochromegane/go-gitignore/
https://github.com/sabhiram/go-gitignore

Of these two, the latter looks more comprehensively implemented, however I do believe it also attempts to match some rules which are specific to gitignore and wouldn't cater for the dockerignore case (I've yet to verify this, but will).

Reply:
This approach sounds reasonable. We have worked with smaller libraries in the past, although ideally we use something with a bit more traction.

I'll also let @garethr & @jpreese have a look at this.

Reply:
I think this is reasonable. Having this is better in a library than in conftest, and if it becomes unsupported, or the ignore formats evolve greatly, then it should be easy enough to swap out.